core/src/test/scala/unit/kafka/log/LogCleanerTest.scala

```diff
@@ -599,12 +599,12 @@ class LogCleanerTest {
     val log = makeLog(config = LogConfig.fromProps(logConfig.originals, logProps))
 
     val appendFirstTransaction = appendTransactionalAsLeader(log, producerId, producerEpoch,
-      origin = AppendOrigin.Replication)
+      origin = AppendOrigin.Coordinator)
```

Comment:
Since now for `AppendOrigin.Replication`, the `LogValidator` is bypassed when `Log.appendAsLeader` is called, which has an assumption that the records to be appended should have leader epoch ready in its underlying batches. And moreover, the transaction/group coordinator are supposed to be calling the `Log.appendAsLeader` with `origin = AppendOrigin.Coordinator`.

Reply:
Reverted since we have a new added `AppendOrigin.RaftLeader`